### DISCUSSION TITLE
How to build Moonray as a DCC plugin

### DISCUSSION THREAD

**Discussion by aaronmack** [dreamworksanimation/openmoonray#125] [General]
Could you give me a detailed guide? To build hdMoonray for houdini and so on.

Much appreciated!

1. I tried to link libhboost_python and all the libpxr_ usd libraries in hfs19.5/dsolib, and  `export TF_DEBUG=*` before start houdinifx, But I got error:

`/home/aaron/openmoonray/release/plugin/hdMoonray.so: undefined symbol: _ZN5boost6python23throw_error_already_setEv`

**Comment by jmahovsky-dwa** [comment]
That symbol comes from boost_python whose name varies depending on the boost version and the python version.  Try setting BOOST_PYTHON_COMPONENT_NAME to the python version you are using.

**Comment by rgwilson68** [comment]
Hi,
We don't have a detailed guide right now, but BrianHanke has written up some info on Houdini builds here: https://github.com/BrianHanke/openmoonray-bhgc/discussions/3

You shouldn't need to build or link hdMoonray with libhboost_python : Moonray can use its own copy of boost, because Houdini's hboost is namespaced and uses a different library name. This mean that regular boost is a different library with different symbols as far as the linker is concerned.

You do, however, have to build and link against Houdini's version of USD, because USD is part of the Hydra interface between Houdini and hdMoonray -- if they use binary incompatible versions the plugin will likely crash. To build hdMoonray with Houdini's USD, you need a CMake config file called pxrConfig.cmake, which unfortunately Houdini does not provide. Here is one I made that seems to work: https://docs.openmoonray.org/assets/build_extras/moonray_houdini_pxr_config.tar

**Comment by rgwilson68** [comment]
See also this discussion: https://github.com/dreamworksanimation/openmoonray/issues/72

**Reply by mpjohn** [reply]
Is there an estimated time for when there will be a detailed guide for this. Brian's method is very helpful but judging by his most recent posts he has struggled to replicate it on Rocky Linux. 
Speaking for myself it's a bit of a sucker punch once you successfully build Moonray to find out that you're only half way there...and there's no proper map for the second leg of the journey.
